### PR TITLE
fix(cli): watch did not copy changed assets referenced by Stylable files

### PR DESCRIPTION
**Pull request by barak007**
fixes: #1923

### CODE REVIEW

File: packages/cli/src/build.ts [hunk 1]
```diff
@@ -121,16 +121,27 @@ export async function build({
             if (!indexFile && outputSources && filePath.startsWith(fullOutDir)) {
                 return false;
             }
+            // assets changes that tracked by stylable file should retrigger processFiles
```
Comment (tomrav) [inline packages/cli/src/build.ts:124]:
```suggestion
            // assets used in stylable files should re-trigger "processFiles" when changed
```

File: packages/cli/src/build.ts [hunk 2]
```diff
@@ -121,16 +121,27 @@ export async function build({
...
+            }
+            // stylable files
             return filePath.endsWith(extension);
         },
         onError(error) {
             console.error(error);
         },
         processFiles(service, affectedFiles, deletedFiles, changeOrigin) {
             if (changeOrigin) {
+                // watched file changed invalidate cache
```
Comment (tomrav) [inline packages/cli/src/build.ts:136]:
```suggestion
                // watched file changed, invalidate cache
```

File: packages/cli/src/build.ts [hunk 3]
```diff
@@ -121,16 +121,27 @@ export async function build({
...
                 stylable.initCache();
+                // handle deleted files by removeing thier generated content
```
Comment (tomrav) [inline packages/cli/src/build.ts:138]:
```suggestion
                // handle deleted files by removing their generated content
```

File: packages/cli/src/build.ts [hunk 4]
```diff
@@ -121,16 +121,27 @@ export async function build({
...
                 if (deletedFiles.size) {
                     for (const deletedFile of deletedFiles) {
+                        if (assets.has(deletedFile)) {
+                            assets.delete(deletedFile);
+                            continue;
+                        }
```
Comment (tomrav) [inline packages/cli/src/build.ts:144]: `else if` for line below?

File: packages/cli/src/build.ts [hunk 5]
```diff
@@ -155,15 +166,27 @@ export async function build({
                 }
             }
 
+            // add files that contains error for retry
```
Comment (tomrav) [inline packages/cli/src/build.ts:169]:
```suggestion
            // add files that contains errors for retry
```

File: packages/cli/src/build.ts [hunk 6]
```diff
@@ -155,15 +166,27 @@ export async function build({
...
+            // remove assets from the affected files (handled in buildAggregatedEntities)
+            for (const filePath of affectedFiles) {
+                if (assets.has(filePath)) {
+                    affectedFiles.delete(filePath);
+                }
+            }
+
+            // rebuild
             buildFiles(affectedFiles);
+            // rewaire invalidations
```
Comment (tomrav) [inline packages/cli/src/build.ts:184]:
```suggestion
            // rewire invalidations
```

File: packages/cli/src/build.ts [hunk 7]
```diff
@@ -155,15 +166,27 @@ export async function build({
...
             updateWatcherDependencies(stylable, service, affectedFiles, sourceFiles);
+            // rebuild assets from agregated content: index files and assets
```
Comment (tomrav) [inline packages/cli/src/build.ts:186]:
```suggestion
            // rebuild assets from aggregated content: index files and assets
```

File: packages/cli/test/cli-watch.spec.ts [hunk 8]
```diff
@@ -121,6 +121,44 @@ describe('Stylable Cli Watch', () => {
...
+        await run({
+            dirPath: tempDir.path,
+            args: ['--outDir', './dist', '-w', '--cjs=false', '--css'],
+            steps: [
+                {
+                    msg: messages.START_WATCHING,
+                    action() {
+                        writeFileSync(
+                            join(tempDir.path, 'asset.svg'),
+                            `<svg height="100" width="100">
```
Comment (tomrav) [inline packages/cli/test/cli-watch.spec.ts:145]: extract to common with fixture above?

File: packages/core/test/stylable-resolver.spec.ts [hunk 9]
```diff
@@ -59,7 +59,6 @@ describe('stylable-resolver', () => {
                 },
             },
         });
-
```
Comment (tomrav) [inline packages/core/test/stylable-resolver.spec.ts:62]: why was this line removed?

**Review by tomrav — CHANGES_REQUESTED** [review]
Needs a couple of fixes

File: packages/core/test/stylable-resolver.spec.ts (hunk 9 above)
Comment (barak007) [reply packages/core/test/stylable-resolver.spec.ts:62]: not sure.

File: packages/cli/src/build.ts (hunk 4 above)
Comment (barak007) [reply packages/cli/src/build.ts]: no need because if the continue...but will do.

**Review by tomrav — APPROVED** [review]

File: packages/cli/test/cli-watch.spec.ts [hunk 10]
```diff
@@ -121,6 +121,39 @@ describe('Stylable Cli Watch', () => {
...
+                        writeFileSync(join(tempDir.path, 'asset.svg'), getSvgContent(NEW_SIZE));
+                    },
+                },
+                {
+                    msg: messages.FINISHED_PROCESSING,
+                },
+            ],
+        });
+        const files = loadDirSync(tempDir.path);
+        expect(files['dist/style.css']).to.include('background:');
```
Comment (tzachbon) [inline packages/cli/test/cli-watch.spec.ts:153]: Why do we need to make sure that `background:` is written?

**Review by tzachbon — APPROVED** [review]
LGTM, left a small question.

File: packages/cli/test/cli-watch.spec.ts (hunk 10 above)
Comment (barak007) [reply packages/cli/test/cli-watch.spec.ts:153]: it's just validating that the content is in tact
